diff --git a/config.rb b/config.rb
--- a/config.rb
+++ b/config.rb
@@ -1,5 +1,5 @@
 # Unique header generation
-require './lib/unique_head.rb'
+require './lib/nesting_unique_head.rb'

 # Markdown
 set :markdown_engine, :redcarpet
@@ -11,7 +11,7 @@ set :markdown,
     tables: true,
     with_toc_data: true,
     no_intra_emphasis: true,
-    renderer: UniqueHeadCounter
+    renderer: NestingUniqueHeadCounter

 # Assets
 set :css_dir, 'stylesheets'
